--- a/googletest/include/gtest/gtest-matchers.h	2021-06-27 19:59:05.693123561 +0900
+++ b/googletest/include/gtest/gtest-matchers.h	2021-06-27 19:59:16.378125646 +0900
@@ -52,6 +52,12 @@
 #define GTEST_MAYBE_5046_
 #endif
 
+#if __GNUG__ && __GNUC__ < 5
+namespace std {
+template <typename T> struct is_trivially_copy_constructible { static constexpr bool value = __has_trivial_copy(T); };
+}
+#endif
+
 GTEST_DISABLE_MSC_WARNINGS_PUSH_(
     4251 GTEST_MAYBE_5046_ /* class A needs to have dll-interface to be used by
                               clients of class B */
