--- a/Makefile.in	2019-07-21 19:32:31.000000000 +0800
+++ b/Makefile.in	2019-09-02 08:40:27.678523255 +0800
@@ -313,7 +313,7 @@
 
 ### Targets: ###
 
-all: $(DLLPREFIX)mysqlpp$(__mysqlpp___targetsuf3) $(LIBPREFIX)mysqlpp_ssqls2parse$(LIBEXT) ssqlsxlat$(EXEEXT) test_array_index$(EXEEXT) test_cpool$(EXEEXT) test_datetime$(EXEEXT) test_inttypes$(EXEEXT) test_insertpolicy$(EXEEXT) test_manip$(EXEEXT) test_null_comparison$(EXEEXT) test_query_copy$(EXEEXT) test_qssqls$(EXEEXT) test_qstream$(EXEEXT) test_sqlstream$(EXEEXT) test_ssqls2$(EXEEXT) test_string$(EXEEXT) test_tcp$(EXEEXT) test_uds$(EXEEXT) test_wnp$(EXEEXT) $(LIBPREFIX)mysqlpp_excommon$(LIBEXT) cgi_jpeg$(EXEEXT) cpool$(EXEEXT) dbinfo$(EXEEXT) deadlock$(EXEEXT) fieldinf$(EXEEXT) for_each$(EXEEXT) load_jpeg$(EXEEXT) multiquery$(EXEEXT) resetdb$(EXEEXT) simple1$(EXEEXT) simple2$(EXEEXT) simple3$(EXEEXT) ssqls1$(EXEEXT) ssqls2$(EXEEXT) ssqls3$(EXEEXT) ssqls4$(EXEEXT) ssqls5$(EXEEXT) ssqls6$(EXEEXT) store_if$(EXEEXT) tquery1$(EXEEXT) tquery2$(EXEEXT) tquery3$(EXEEXT) tquery4$(EXEEXT) transaction$(EXEEXT)
+all: $(DLLPREFIX)mysqlpp$(__mysqlpp___targetsuf3) $(LIBPREFIX)mysqlpp$(LIBEXT) $(LIBPREFIX)mysqlpp_ssqls2parse$(LIBEXT) ssqlsxlat$(EXEEXT) test_array_index$(EXEEXT) test_cpool$(EXEEXT) test_datetime$(EXEEXT) test_inttypes$(EXEEXT) test_insertpolicy$(EXEEXT) test_manip$(EXEEXT) test_null_comparison$(EXEEXT) test_query_copy$(EXEEXT) test_qssqls$(EXEEXT) test_qstream$(EXEEXT) test_sqlstream$(EXEEXT) test_ssqls2$(EXEEXT) test_string$(EXEEXT) test_tcp$(EXEEXT) test_uds$(EXEEXT) test_wnp$(EXEEXT) $(LIBPREFIX)mysqlpp_excommon$(LIBEXT) cgi_jpeg$(EXEEXT) cpool$(EXEEXT) dbinfo$(EXEEXT) deadlock$(EXEEXT) fieldinf$(EXEEXT) for_each$(EXEEXT) load_jpeg$(EXEEXT) multiquery$(EXEEXT) resetdb$(EXEEXT) simple1$(EXEEXT) simple2$(EXEEXT) simple3$(EXEEXT) ssqls1$(EXEEXT) ssqls2$(EXEEXT) ssqls3$(EXEEXT) ssqls4$(EXEEXT) ssqls5$(EXEEXT) ssqls6$(EXEEXT) store_if$(EXEEXT) tquery1$(EXEEXT) tquery2$(EXEEXT) tquery3$(EXEEXT) tquery4$(EXEEXT) transaction$(EXEEXT)
 
 install: install_mysqlpp
 	$(INSTALL_DIR) $(DESTDIR)$(prefix)/include/mysql++
@@ -331,6 +331,7 @@
 	rm -rf ./.deps ./.pch
 	rm -f ./*.o
 	rm -f $(DLLPREFIX)mysqlpp$(__mysqlpp___targetsuf3)
+	rm -f $(LIBPREFIX)mysqlpp$(LIBEXT)
 	rm -f $(LIBPREFIX)mysqlpp.$(DLLIMP_SUFFIX)
 	rm -f $(LIBPREFIX)mysqlpp.$(DLLIMP_SUFFIX) $(DLLPREFIX)mysqlpp$(__mysqlpp___targetsuf2)
 	rm -f $(LIBPREFIX)mysqlpp_ssqls2parse$(LIBEXT)
@@ -391,10 +392,16 @@
 	
 	$(__mysqlpp___so_symlinks_cmd)
 
-install_mysqlpp: $(DLLPREFIX)mysqlpp$(__mysqlpp___targetsuf3)
+$(LIBPREFIX)mysqlpp$(LIBEXT): $(MYSQLPP_OBJECTS) lib/ssqls.h lib/querydef.h
+	rm -f $@
+	$(AR) $(AROPTIONS) $@ $(MYSQLPP_OBJECTS)
+	$(RANLIB) $@
+	
+install_mysqlpp: $(DLLPREFIX)mysqlpp$(__mysqlpp___targetsuf3) $(LIBPREFIX)mysqlpp$(LIBEXT)
 	$(INSTALL_DIR) $(DESTDIR)$(libdir)
 	$(INSTALL_DATA) $(LIBPREFIX)mysqlpp.$(DLLIMP_SUFFIX) $(DESTDIR)$(libdir)
 	$(INSTALL_PROGRAM) $(DLLPREFIX)mysqlpp$(__mysqlpp___targetsuf3) $(DESTDIR)$(libdir)
+	$(INSTALL_PROGRAM) $(LIBPREFIX)mysqlpp$(LIBEXT) $(DESTDIR)$(libdir)
 	(cd $(DESTDIR)$(libdir) ; $(__mysqlpp___so_symlinks_inst_cmd))
 
 uninstall_mysqlpp: 
--- a/lib/common.h	2014-01-24 19:49:33.000000000 +0800
+++ b/lib/common.h	2014-01-24 19:50:29.000000000 +0800
@@ -35,6 +35,9 @@
 #if !defined(DOXYGEN_IGNORE)
 // Doxygen will not generate documentation for the following stuff.
 
+// Bury MySQL headers by default.
+#define MYSQLPP_MYSQL_HEADERS_BURIED
+
 // Enable SSQLS by default.  Turned off below on platforms where we
 // know it doesn't work.
 #define MYSQLPP_SSQLS_COMPATIBLE
