--- a/include/my_alloc.h	2016-03-29 02:06:12.000000000 +0800
+++ b/include/my_alloc.h	2016-05-05 14:45:43.000000000 +0800
@@ -24,7 +24,7 @@
 #define ALLOC_MAX_BLOCK_USAGE_BEFORE_DROP	10
 
 /* PSI_memory_key */
-#include <mysql/psi/psi_memory.h>
+#include "mysql/psi/psi_memory.h"
 
 #ifdef __cplusplus
 extern "C" {
--- a/include/mysql.h.pp	2016-03-29 02:06:12.000000000 +0800
+++ b/include/mysql.h.pp	2016-05-09 22:39:53.000000000 +0800
@@ -216,7 +216,7 @@
   int type; unsigned int interface_version; const char *name; const char *author; const char *desc; unsigned int version[3]; const char *license; void *mysql_api; int (*init)(char *, size_t, int, va_list); int (*deinit)(); int (*options)(const char *option, const void *);
 };
 struct st_mysql;
-#include <mysql/plugin_auth_common.h>
+#include "plugin_auth_common.h"
 typedef struct st_plugin_vio_info
 {
   enum { MYSQL_VIO_INVALID, MYSQL_VIO_TCP, MYSQL_VIO_SOCKET,
@@ -280,7 +280,7 @@
 typedef unsigned long long my_ulonglong;
 #include "typelib.h"
 #include "my_alloc.h"
-#include <mysql/psi/psi_memory.h>
+#include "mysql/psi/psi_memory.h"
 #include "psi_base.h"
 struct PSI_thread;
 typedef unsigned int PSI_memory_key;
--- a/include/mysql/client_plugin.h	2016-05-09 22:40:52.000000000 +0800
+++ b/include/mysql/client_plugin.h	2016-05-09 22:41:14.000000000 +0800
@@ -100,7 +100,7 @@
 struct st_mysql;
 
 /******** authentication plugin specific declarations *********/
-#include <mysql/plugin_auth_common.h>
+#include "plugin_auth_common.h"
 
 struct st_mysql_client_plugin_AUTHENTICATION
 {
--- a/include/mysql/client_plugin.h.pp	2016-05-09 22:41:01.000000000 +0800
+++ b/include/mysql/client_plugin.h.pp	2016-05-09 22:41:29.000000000 +0800
@@ -3,7 +3,7 @@
   int type; unsigned int interface_version; const char *name; const char *author; const char *desc; unsigned int version[3]; const char *license; void *mysql_api; int (*init)(char *, size_t, int, va_list); int (*deinit)(); int (*options)(const char *option, const void *);
 };
 struct st_mysql;
-#include <mysql/plugin_auth_common.h>
+#include "plugin_auth_common.h"
 typedef struct st_plugin_vio_info
 {
   enum { MYSQL_VIO_INVALID, MYSQL_VIO_TCP, MYSQL_VIO_SOCKET,
