--- a/tinyxml.cpp	2011-05-15 10:24:57.000000000 +0800
+++ b/tinyxml.cpp	2014-01-26 15:05:17.000000000 +0800
@@ -57,30 +57,7 @@
 	{
 		unsigned char c = (unsigned char) str[i];
 
-		if (    c == '&' 
-		     && i < ( (int)str.length() - 2 )
-			 && str[i+1] == '#'
-			 && str[i+2] == 'x' )
-		{
-			// Hexadecimal character reference.
-			// Pass through unchanged.
-			// &#xA9;	-- copyright symbol, for example.
-			//
-			// The -1 is a bug fix from Rob Laveaux. It keeps
-			// an overflow from happening if there is no ';'.
-			// There are actually 2 ways to exit this loop -
-			// while fails (error case) and break (semicolon found).
-			// However, there is no mechanism (currently) for
-			// this function to return an error.
-			while ( i<(int)str.length()-1 )
-			{
-				outString->append( str.c_str() + i, 1 );
-				++i;
-				if ( str[i] == ';' )
-					break;
-			}
-		}
-		else if ( c == '&' )
+		if ( c == '&' )
 		{
 			outString->append( entity[0].str, entity[0].strLength );
 			++i;
--- a/tinyxml.h	2011-05-15 10:24:57.000000000 +0800
+++ b/tinyxml.h	2014-01-26 15:05:17.000000000 +0800
@@ -26,6 +26,10 @@
 #ifndef TINYXML_INCLUDED
 #define TINYXML_INCLUDED
 
+#ifndef TIXML_USE_STL
+#define TIXML_USE_STL
+#endif
+
 #ifdef _MSC_VER
 #pragma warning( push )
 #pragma warning( disable : 4530 )
--- a/xmltest.cpp	2011-05-15 10:24:57.000000000 +0800
+++ b/xmltest.cpp	2014-01-26 15:05:17.000000000 +0800
@@ -3,6 +3,10 @@
 */
 
 
+#ifndef TIXML_USE_STL
+#define TIXML_USE_STL
+#endif
+
 #ifdef TIXML_USE_STL
 	#include <iostream>
 	#include <sstream>
--- a/Makefile	2011-05-15 10:24:57.000000000 +0800
+++ b/Makefile	2014-04-26 00:07:11.000000000 +0800
@@ -81,37 +81,68 @@
 # Targets of the build
 #****************************************************************************
 
-OUTPUT := xmltest
+SYSTEM := $(shell uname -s)
+ifeq ($(SYSTEM),Darwin)
+SONAME := libtinyxml.0.0.0.dylib
+else
+SONAME := libtinyxml.so.0.0.0
+endif
 
-all: ${OUTPUT}
+OUTPUT := xmltest libtinyxml.a $(SONAME)
 
+ifeq ($(SYSTEM),Darwin)
+LDFLAGS_SO := -install_name `cd ../../.. && pwd`/staging/lib/libtinyxml.0.dylib
+all: ${OUTPUT}
+	ln -sf $(SONAME) libtinyxml.0.dylib
+	ln -sf $(SONAME) libtinyxml.dylib
+else
+all: ${OUTPUT}
+	ln -sf $(SONAME) libtinyxml.so.0
+	ln -sf $(SONAME) libtinyxml.so
+endif
 
 #****************************************************************************
 # Source files
 #****************************************************************************
 
-SRCS := tinyxml.cpp tinyxmlparser.cpp xmltest.cpp tinyxmlerror.cpp tinystr.cpp
+SRCS := tinyxml.cpp tinyxmlparser.cpp xmltest.cpp tinyxmlerror.cpp
+LIBSRCS := tinyxml.cpp tinyxmlparser.cpp tinyxmlerror.cpp
 
 # Add on the sources for libraries
 SRCS := ${SRCS}
+LIBSRCS := ${LIBSRCS}
 
 OBJS := $(addsuffix .o,$(basename ${SRCS}))
+LIBOBJS := $(addsuffix .o,$(basename ${LIBSRCS}))
+LIBSHOBJS := $(addsuffix .sho,$(basename ${LIBSRCS}))
 
 #****************************************************************************
 # Output
 #****************************************************************************
 
-${OUTPUT}: ${OBJS}
+xmltest: ${OBJS}
 	${LD} -o $@ ${LDFLAGS} ${OBJS} ${LIBS} ${EXTRA_LIBS}
 
+libtinyxml.a: ${LIBOBJS}
+	${AR} $@ ${LIBOBJS}
+
+$(SONAME): ${LIBSHOBJS}
+	${CXX} ${LDFLAGS} ${LDFLAGS_SO} -shared -o $@ ${LIBSHOBJS}
+
 #****************************************************************************
 # common rules
 #****************************************************************************
 
 # Rules for compiling source files to object files
+%.sho : %.cpp
+	${CXX} -c -fPIC ${CXXFLAGS} ${INCS} $< -o $@
+
 %.o : %.cpp
 	${CXX} -c ${CXXFLAGS} ${INCS} $< -o $@
 
+%.sho : %.c
+	${CC} -c -fPIC ${CFLAGS} ${INCS} $< -o $@
+
 %.o : %.c
 	${CC} -c ${CFLAGS} ${INCS} $< -o $@
 
@@ -119,7 +150,7 @@
 	bash makedistlinux
 
 clean:
-	-rm -f core ${OBJS} ${OUTPUT}
+	-rm -f core *.o *.sho libtinyxml.a libtinyxml.so* xmltest
 
 depend:
 	#makedepend ${INCS} ${SRCS}
