--- a/src/xlslib/workbook.h	2013-11-15 18:44:47.000000000 +0800
+++ b/src/xlslib/workbook.h	2013-11-15 18:50:28.000000000 +0800
@@ -43,8 +43,6 @@
 
 // #include "common/xls_pshpack2.h"
 
-#define XLSLIB_VERSION PACKAGE_VERSION
-
 namespace xlslib_core
 {
 	typedef enum
@@ -65,7 +63,7 @@
 		workbook();
 		virtual ~workbook();
 
-		const char*	version() const { return XLSLIB_VERSION; }
+		const char*	version() const;
 
 		worksheet*	sheet(const std::string& sheetname);
 		worksheet*	sheet(const xlslib_strings::ustring& sheetname);
--- a/src/xlslib/workbook.cpp	2014-11-03 07:10:50.000000000 +0800
+++ a/src/xlslib/workbook.cpp	2015-04-07 17:22:39.000000000 +0800
@@ -52,6 +52,13 @@
 using namespace xlslib_core;
 using namespace xlslib_strings;
 
+/* Don't pollute global namespace. */
+#include "common/xlconfig.h"
+#define XLSLIB_VERSION PACKAGE_VERSION
+const char*	workbook::version() const
+{
+    return XLSLIB_VERSION;
+}
 
 #define CHANGE_DUMPSTATE(state) {				\
 		m_PreviousDumpState = m_DumpState;			 \
