--- a/src/zookeeper.c	2014-11-04 15:38:25.000000000 +0800
+++ b/src/zookeeper.c	2014-11-04 15:38:42.000000000 +0800
@@ -426,6 +426,7 @@
     } else {
         int rc = read(fd, &seed, sizeof(seed));
         assert(rc == sizeof(seed));
+        (void)rc;
         close(fd);
     }
     srandom(seed);
--- a/src/zk_hashtable.c	2014-11-04 15:50:22.000000000 +0800
+++ b/src/zk_hashtable.c	2014-11-04 15:50:39.000000000 +0800
@@ -188,6 +188,7 @@
         /* inserting a new path element */
         res=hashtable_insert(ht->ht,strdup(path),create_watcher_object_list(wo));
         assert(res);
+        (void)res;
     }else{
         /*
          * Path already exists; check if the watcher already exists.
